server/auth_server.go

```diff
@@ -370,6 +370,22 @@ func updateSettings(saveConfig map[string]map[string]string) error {
 		// authConfig now was prevConfig
 		// saveConfig is to be saved, so authConfig should get added values from saveConfig
 		log.Debugf("Previous saved auth config: %v", authConfig)
+		// saveConfig (updated config) might contain no secret settings, if GetConfig(listOnly=true) was called before it
```

Comment:
@prachidamle The POST to v1-auth/config when changing access mode does not have the secret settings because it uses the config it gets from GetConfig(listOnly=true).
This was the same earlier also for github and ad before settings were encrypted.
But the updateSettings function just saved the new settings to db and didn't touch any of the secret settings then. So the secret setting was still intact in the setting table
Now that its a single GO, the updateSettings encrypts the config that does not contain secretSettings and then the secrets are lost.